CMake/RealmCore.cmake

```diff
@@ -120,6 +128,13 @@ function(download_realm_tarball url target libraries)
             COMMAND ${CMAKE_COMMAND} -E make_directory ${target}
             COMMAND ${CMAKE_COMMAND} -E chdir ${target} tar xf ${tarball_path}
             COMMAND ${CMAKE_COMMAND} -E touch_nocreate ${libraries})
+    elseif(CMAKE_SYSTEM_NAME MATCHES "^Windows")
+        add_custom_command(
+            COMMENT "Extracting ${tarball_name}"
+            OUTPUT ${libraries}
+            COMMAND "${CMAKE_COMMAND}" -E make_directory "${target}"
```

Comment:
If I'm reading this right, it could be combined with the Android case as the only difference is whether `tar` is invoked directly vs via a CMake helper. The CMake helper should work fine for Windows as well.